Utils/azuretools-core/src/com/microsoft/azuretools/azurecommons/util/GetHashMac.java

```diff
@@ -50,22 +57,21 @@ public static boolean IsValidHashMacFormat(String hashMac) {
 
         Pattern hashmac_pattern = Pattern.compile("[0-9a-f]{64}");
         Matcher matcher = hashmac_pattern.matcher(hashMac);
-        return matcher.matches();
+        return matcher.matches() && !(StringUtils.containsIgnoreCase(System.getProperty("os.name"), "mac") &&
+                StringUtils.equalsIgnoreCase(INVALIDATE_HASHED_MAC_ADDRESS, hashMac));
```

Comment:
add a new dedicated method for isInstallationIdDepreciated, otherwise the logic in IsValidHashMacFormat is very odd.

Reply:
Added `IsValidHashMac`, which calls `IsValidHashMacFormat` and `isInstallationIdDepreciated`